source/features/show-whitespace.tsx

```diff
@@ -21,26 +21,38 @@ function showWhiteSpacesOn(line: Element): void {
 			continue;
 		}
 
-		const startingCharacter = shouldAvoidSurroundingSpaces && nodeIndex === 0 ? 1 : 0;
-		const skipLastCharacter = shouldAvoidSurroundingSpaces && nodeIndex === textNodesOnThisLine.length - 1;
+		// This referes to the boundary text nodes, not actual whitespace nodes. Text nodes are generated by the syntax highlighter, so non-highlighted text will have one text node.
```

Comment:
Not that this PR doesn't change how the feature behaves on non-highlighted code like

```
hello world
```